onnxruntime/core/providers/webgpu/webgpu_context.h

```diff
@@ -34,26 +34,51 @@ struct CapturedCommandInfo {
   WGPUBuffer indirect_buffer;  // WGPUBuffer for indirect dispatch, nullptr if not using indirect dispatch
 };
 
-struct WebGpuContextConfig {
-  int context_id;
-  WGPUInstance instance;
-  WGPUDevice device;
-  const void* dawn_proc_table;
-  ValidationMode validation_mode;
-  bool preserve_device;
-  uint64_t max_storage_buffer_binding_size;
-  int power_preference;
-};
-
 struct WebGpuBufferCacheConfig {
   struct ConfigEntry {
     BufferCacheMode mode;
-    std::string config_string;
+    std::string config_string;  // preserved for customized configuration, eg. bucket sizes
+  };
+  ConfigEntry storage{BufferCacheMode::Bucket, {}};
+  ConfigEntry uniform{BufferCacheMode::Simple, {}};
+  ConfigEntry query_resolve{BufferCacheMode::Disabled, {}};
+  ConfigEntry default_entry{BufferCacheMode::Disabled, {}};
+};
+
+/// <summary>
+/// Represents the configuration options for creating a WebGpuContext.
+/// </summary>
+struct WebGpuContextConfig {
+  int context_id{0};
+  WGPUInstance instance{nullptr};
+  WGPUDevice device{nullptr};
+  const void* dawn_proc_table{nullptr};
+  ValidationMode validation_mode{
+#ifndef NDEBUG
+      webgpu::ValidationMode::Full  // for debug build, enable full validation by default
+#else
+      webgpu::ValidationMode::Basic  // for release build, enable basic validation by default
+#endif  // !NDEBUG
   };
-  ConfigEntry storage;
-  ConfigEntry uniform;
-  ConfigEntry query_resolve;
-  ConfigEntry default_entry;
+  bool preserve_device{false};
+  uint64_t max_storage_buffer_binding_size{0};
+  WebGpuBufferCacheConfig buffer_cache_config{};
```

Comment:
`buffer_cache_config` isn’t required for creating a WebGPU context, but it is shared across sessions. From a functionality perspective, do you think it makes more sense to define `buffer_cache_config` as a session option, or should we keep it as a WebGPU‑context option since all sessions share it?

Reply:
The current implementation of the buffer cache is per-WebGpuContext, so it makes sense to me that the corresponding config is per-WebGpuContext.

We can make some optimization to the buffer cache to further optimize GPU memory usage. But this is going to be a future work item and I am not including it in this PR.